### PR TITLE
refactor: remove old stype strategy support [BREAKING CHANGE]

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

- drop node < 8 support
- drop  old style strategy support
- change to async, but still support generator
- use egg@2

### CODE REVIEW

File: agent.js [hunk 1]
```diff
@@ -9,37 +8,13 @@ module.exports = agent => {
...
-      return handlers;
-    },
-  });
-
+  // wait for other plugin to register custom strategy
   agent.beforeStart(() => {
-    // wait for other plugin to register custom strategy
-    const keys = Object.keys(handlers);
-    for (const type of keys) {
-      agent.schedule.use(type, handler2Class(type, handlers[type]));
```
Comment (atian25) [inline agent.js]: 主要变更点1：移除旧的扩展方式

File: lib/load_schedule.js [hunk 2]
```diff
@@ -30,12 +29,13 @@ function getScheduleLoader(app) {
...
+          task = ctx => {
             const s = new schedule(ctx);
-            yield s.subscribe();
-          });
+            s.subscribe = app.toAsyncFunction(s.subscribe);
+            return s.subscribe();
+          };
         } else {
-          task = co.wrap(schedule.task);
+          task = app.toAsyncFunction(schedule.task);
```
Comment (atian25) [inline lib/load_schedule.js]: 主要变更2：async first

File: test/fixtures/async/app/schedule/sub/cron.js [hunk 3]
```diff
@@ -1,23 +1,12 @@
...
 
-var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
-  return new (P || (P = Promise))(function (resolve, reject) {
-    function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
-    function rejected(value) { try { step(generator.throw(value)); } catch (e) { reject(e); } }
-    function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
-    step((generator = generator.apply(thisArg, _arguments)).next());
-  });
-};
-
```
Comment (atian25) [inline test/fixtures/async/app/schedule/sub/cron.js]: `__awaiter` 用例不用保留了吧？

**Comment by atian25** [comment]
@popomore 看看

**Comment by dead-horse** [comment]
3.0.0

**Pull request merged**